2017/proposals/ranasinghe-kanchana.md

```diff
@@ -0,0 +1,104 @@
+
+| **Intensity** | **Involves**  | **Mentors** |
+| ------------- | --------------|------------ |
+| Intermediate | Python, Data science | [@tacaswell][@story645] |
+
+## Technical Details
+With the widespread presence and usage of large datasets of varied nature, the analysis of categorical data has received much attention. Most machine learning, big-data analysis, and data-mining research and projects require the datasets available to be visualized at various stages of work. A variety of plotting techniques are utilized for this matter, and Matplotlib plays a prominent role as an essential tool for many researchers. Despite its diverse functionality with regard to numerical data visualization, Matplotlib is quite restricted in the domain of categorical data analysis. In most cases, user are compelled to manually map their categorical data to numerical data, thus introducing unnecessary relationships and patterns to the datasets, creating redundant data, and being withheld from the simplicity and ease of usage inherent to Matplotlib. 
+This work hopes to tackle this issue to a certain extent. The Categorical Color Project will include two main tasks: building the heat-map and other scalarMappables API, and normalization and color-map APIs. Matplotlib currently has these frameworks set-up with functionality for numerical data. This functionality will be extended to support categorical data. The work of this project will be done solely using Python. Libraries used will include Numpy and Matplotlib. There will be no new (unused in Matplotlib previously) third-party libraries used in the course of this project. 
+The work of this project will stem from existing development done categorical color support in Matplotlib issues [6889](https://github.com/matplotlib/matplotlib/pull/6889) and [7383](https://github.com/matplotlib/matplotlib/issues/7383). The starting point for the project will be extending the imshow function (matplotlib.axes._axes.imshow) to support categorical data. This mainly focuses on providing it unit knowledge with regards to non-numerical data. The approach followed here will be the initial basis for this extended support for categorical data. It will be used as a foundation to build the heat-maps API, and normalization and color-map APIs. 
+Currently, the process of using matplotlib.pyplot for heat-maps and other scalarMappables includes the creation of a color-map followed by normalization of all available data to fit this color-map. Color-maps plotting functionality is threefold: sequential, divergent and qualitative. Considering the intrinsic discreteness of categorical data, the extension of the color-maps API will focus on its qualitative plotting ability. However the fundamental behavior of the color-maps API will require changes for handling categorical data as it is currently dependent on the presence of numerical scalar values in the data to be plotted. 
```

Comment:
Not exactly - the user chooses a colormap and a normalization to use with that colormap. And this outright isn't true (and again, should be avoided at all costs:)
```
However the fundamental behavior of the color-maps API will require changes for handling categorical data as it is currently dependent on the presence of numerical scalar values in the data to be plotted. 
```